File: zoho_fetch.py
import requests
import json
from zoho_token_manager import get_headers, get_api_domain

organization_id = "892673756"
api_domain = get_api_domain()
headers = get_headers(organization_id)

# Endpoint for invoices
url = f"{api_domain}/books/v3/invoices"

try:
    response = requests.get(url, headers=headers)
    response.raise_for_status()  # Raise HTTPError for bad responses
    invoices = response.json()
    print("Successfully fetched invoices.")
    print(json.dumps(invoices, indent=2))
except requests.exceptions.RequestException as e:
    print(f"Error fetching invoices: {e}")
    if response is not None:
        print(f"Response status: {response.status_code}")
        print(f"Response body: {response.text}")

File: zoho_fetch.js
const axios = require('axios');
const { getHeaders, getApiDomain } = require('./zoho_token_manager');

async function fetchInvoices() {
    const organizationId = "892673756";
    const apiDomain = getApiDomain();
    const headers = await getHeaders(organizationId);

    // Endpoint for invoices
    const url = `${apiDomain}/books/v3/invoices`;

    try {
        const response = await axios.get(url, { headers });
        const invoices = response.data;
        console.log("Successfully fetched invoices.");
        console.log(JSON.stringify(invoices, null, 2));
        return invoices;
    } catch (error) {
        console.log(`Error fetching invoices: ${error.message}`);
        if (error.response) {
            console.log(`Response status: ${error.response.status}`);
            console.log(`Response body: ${error.response.data}`);
        }
        throw error;
    }
}

// Execute the function if this file is run directly
if (require.main === module) {
    fetchInvoices();
}

module.exports = { fetchInvoices };